homework/sophie_hw/metaphlan2_notes.md

```diff
@@ -0,0 +1,336 @@
+`OSError: Not a gzipped file (b'>H')`
+* Check .txt file, empty
+
+`$ less -S SRS014459-Stool_profile.txt`
+
+* Prof Google not helpful
+* Sent question to Lauren Tso & Kevin Bonham
+
+### Error trouble shooting
+1. *From Kevin* - Click download, rename to not have .gz, then zip
+
+`$ mv ~/Downloads/SRS*.fasta.gz ~/gitrepos/metaphlan_tutorial/`
+
+`$ mv SRS014476-Supragingival_plaque.fasta.gz SRS014476-Supragingival_plaque.fasta`
+
+`$ gzip SRS014476-Supragingival_plaque.fasta`
+
+`$ metaphlan2.py SRS014476-Supragingival_plaque.fasta.gz --input_type fasta --nproc 4 > SRS014459-Supragingival_plaque_profile.txt`
+
+2. *From Lauren* Delete all files, re-download everything using `$ curl -O`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014476-Supragingival_plaque.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014494-Posterior_fornix.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014459-Stool.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014464-Anterior_nares.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014470-Tongue_dorsum.fasta.gz`
+
+`$ curl -O https://bitbucket.org/biobakery/biobakery/raw/tip/demos/biobakery_demos/data/metaphlan2/input/SRS014472-Buccal_mucosa.fasta.gz`
+
+`$ mv ~/SRS*.fasta.gz ~/gitrepos/metaphlan_tutorial`
+*Only necessary if dir not ~/gitrepos/metaphlan_tutorial*
+
+## Run multiple Samples
+`$ metaphlan2.py SRS014464-Anterior_nares.fasta.gz --input_type fasta --nproc 4 > SRS014464-Anterior_nares_profile.txt`  
+
+`$ metaphlan2.py SRS014470-Tongue_dorsum.fasta.gz --input_type fasta --nproc 4 > SRS014470-Tongue_dorsum_profile.txt`
+
+`$ metaphlan2.py SRS014472-Buccal_mucosa.fasta.gz --input_type fasta --nproc 4 > SRS014472-Buccal_mucosa_profile.txt`
+
+`$ metaphlan2.py SRS014494-Posterior_fornix.fasta.gz --input_type fasta --nproc 4 > SRS014494-Posterior_fornix_profile.txt`
+
+* Create single tab-delimited table from output files
+`$ merge_metaphlan_tables.py *_profile.txt > merged_abundance_table.txt`
+
+`$ less -S merged_abundance_table.txt` :q
+
+# Vistualiza results
+
+## Create a heatmap with hclust2
+* hclust2 = plotting tool
+
+#### Install hclust2
+`$ brew tap biobakery/biobakery`
+
+`$ brew install hclust2`
+
+### Step 1: Generate the species only abundance table
+`$ grep -E "(s__)|(^ID)" merged_abundance_table.txt | grep -v "t__" | sed 's/^.*s__//g' > merged_abundance_table_species.txt`
+
+`$ less -S merged_abundance_table_species.txt` :q
+
+#### What's happening in this code?
+* grep = finds specified patterns within text, "global regular expression print"
+1. grep 1: Searches for `"(s__)|(^ID)"` = lines with species info & also to the header
```

Comment:
Regular expressions are super powerful, and definitely worth learning. Check out regexr.com for cheatsheets and to practice/test your regular expressions. I still use this site a ton.